lib/content-services/document-list/components/library-name-column/library-name-column.component.ts

```diff
@@ -16,75 +16,105 @@
  */
 
 import {
-  Component,
-  ChangeDetectionStrategy,
-  ViewEncapsulation,
-  OnInit,
-  Input,
-  ElementRef
+    Component,
+    ChangeDetectionStrategy,
+    ViewEncapsulation,
+    OnInit,
+    Input,
+    ElementRef,
+    OnDestroy
 } from '@angular/core';
-import { NodeEntry } from '@alfresco/js-api';
+import { NodeEntry, Node } from '@alfresco/js-api';
 import { ShareDataRow } from '../../data/share-data-row.model';
+import { AlfrescoApiService } from '@alfresco/adf-core';
+import { BehaviorSubject, Subscription } from 'rxjs';
 
 @Component({
-  selector: 'adf-library-name-column',
-  template: `
-    <span title="{{ displayTooltip }}" (click)="onClick()">
-      {{ displayText }}
-    </span>
-  `,
-  changeDetection: ChangeDetectionStrategy.OnPush,
-  encapsulation: ViewEncapsulation.None,
-  host: { class: 'adf-datatable-cell adf-datatable-link adf-library-name-column' }
+    selector: 'adf-library-name-column',
+    template: `
+        <span title="{{ displayTooltip$ | async }}" (click)="onClick()">
+            {{ displayText$ | async }}
+        </span>
+    `,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    host: {
+        class: 'adf-datatable-cell adf-datatable-link adf-library-name-column'
+    }
 })
-export class LibraryNameColumnComponent implements OnInit {
-  @Input()
-  context: any;
-
-  displayTooltip: string;
-  displayText: string;
-  node: NodeEntry;
-
-  constructor(private element: ElementRef) {}
-
-  ngOnInit() {
-    this.node = this.context.row.node;
-    const rows: Array<ShareDataRow> = this.context.data.rows || [];
-    if (this.node && this.node.entry) {
-      this.displayText = this.makeLibraryTitle(this.node.entry, rows);
-      this.displayTooltip = this.makeLibraryTooltip(this.node.entry);
+export class LibraryNameColumnComponent implements OnInit, OnDestroy {
+    @Input()
+    context: any;
+
+    displayTooltip$ = new BehaviorSubject<string>('');
+    displayText$ = new BehaviorSubject<string>('');
+    node: NodeEntry;
+
+    private sub: Subscription;
+
+    constructor(private element: ElementRef, private api: AlfrescoApiService) {}
```

Comment:
would it better as we discuss rename api to alfrescoApiService? it helps consistency